scenes/autoload/data_manager.gd

```diff
@@ -16,79 +16,79 @@ var pending_file_changes :Dictionary = {}
 ## save changes to disk at the end of current update frame
 var auto_apply_changes := true
 
-class SectionWrapper:
-	var file_path:String
-	var section:String
-	func set_data(key: String, value: Variant):
-		DataManager.set_data(file_path, section, key, value)
-	func get_data(key: String, default: Variant = null) -> Variant:
-		return DataManager.get_data(file_path, section, key, default)
-	func get_keys() -> PackedStringArray:
-		return DataManager.get_config(file_path).get_section_keys(section)
-
 func _ready() -> void:
 	for file_path in LOAD_ON_READY_PATHS:
-		get_config(file_path)
+		_get_config(file_path)
 
 func get_wrapper(file_path: String, section: String) -> SectionWrapper:
 	var wrapper := SectionWrapper.new()
 	wrapper.file_path = file_path
 	wrapper.section = section
 	return wrapper
 
-func get_config(file_path: String) -> ConfigFile:
-	if not file_data.has(file_path):
-		file_data[file_path] = load_file(file_path)
-	return file_data[file_path]
+func get_data(file_path: String, section: String, key :String, default :Variant = null) -> Variant:
+	var config := _get_config(file_path)
+	return config.get_value(section, key, default)
 
 func set_data(file_path: String, section: String, key: String, value: Variant):
-	var config := get_config(file_path)
+	var config := _get_config(file_path)
 	config.set_value(section, key, value)
 	pending_file_changes[file_path] = true
-	if auto_apply_changes and not get_tree().process_frame.is_connected(save_pending):
-		get_tree().process_frame.connect(save_pending, CONNECT_ONE_SHOT)
+	if auto_apply_changes and not get_tree().process_frame.is_connected(_save_pending):
+		get_tree().process_frame.connect(_save_pending, CONNECT_ONE_SHOT)
 
-func get_data(file_path: String, section: String, key :String, default :Variant = null) -> Variant:
-	var config := get_config(file_path)
-	return config.get_value(section, key, default)
+func save_high_score(key: String, value) -> void:
+	if is_high_score(key, value):
+		set_data(HIGHSCORE_FILE_PATH, HIGHSCORE_SECTION, key, value)
+
+func is_high_score(key: String, value) -> bool:
+	return value > get_high_score(key)
+
+func get_high_score(key: String) -> int:
+	return get_data(HIGHSCORE_FILE_PATH, HIGHSCORE_SECTION, key, 0)
+
+func _get_config(file_path: String) -> ConfigFile:
+	if not file_data.has(file_path):
+		file_data[file_path] = _load_file(file_path)
+	return file_data[file_path]
 
 # there's no reason to use this, but it's good to have it for completeness
 func save_all_data() -> void:
 	for file_path in file_data:
-		save_file(file_path)
+		_save_file(file_path)
 	pending_file_changes.clear()
 
-func save_pending():
+func _save_pending():
```

Comment:
Actually save_pending and cancel_pending were intended as public functions, not private ones

you can potentially disable auto_apply_changes to stop it from writing to disk
and then use save_pending() and check auto back to true when you are done changing

Reply:
Then let's chang it back to public